src/content/reference/react/cache.md

```diff
@@ -0,0 +1,443 @@
+
+</DeepDive>
+
+<DeepDive>
+
+#### When should I not use cache? {/*not-use-cache*/}
+
+`cache` is not recommended for use in Client Components today.
+
+The motivation for `cache` is to provide a complete data fetching and invalidation strategy between client and server. However, only server usage is ready for use today.
+
+In a React Server Component environment, `cache` is most useful for providing data snapshots or memoizing expensive computations across the render tree.
+
+[Deterministic](https://en.wikipedia.org/wiki/Deterministic_algorithm) functions are good candidates for `cache`. However, if there are side-effects that you want to run for each invocation, then it is not a good option for `cache`.
+
+Although data-fetching is often not deterministic, as there may be mutations between fetches, the assumption is that fetched data should be consistent across a render. This makes data-fetching logic well-suited to `cache`. As the cache is invalidated across server requests, your app will still fetch fresh data in the next request. However, this may not fit for your app's data-fetching use-case and `cache` may not useful.
+
+</DeepDive>
+
+---
+
+## Troubleshooting {/*troubleshooting*/}
+
+### My memoized function still runs even though I've called it with the same arguments {/*memoized-function-still-runs*/}
+
+When calling a memoized function, React will look up the input arguments to see if a result is already cached. React will use shallow equality of the arguments to determine if there is a cache hit.
+
+This means that if your arguments are not [primatives](https://developer.mozilla.org/en-US/docs/Glossary/Primitive) (ex. objects, functions, arrays), ensure you're passing the same object reference.
+
+```js
+import {cache} from 'react';
+
+const calculateNorm = cache((vector) => {
+  // ...
+});
+
+function MapMarker(props) {
+  // 🚩 Wrong: props is an object that changes every render.
+  const length = calculateNorm(props);
+  // ...
+}
+
+function App() {
+  return (
+    <>
+      <MapMarker x={10} y={10} z={10} />
+      <MapMarker x={10} y={10} z={10} />
+    </>
+  );
+}
+```
+
+In this case the two `MapMarker`s look like they're doing the same work and calling `calculateNorm` with the same value of `{x: 10, y: 10, z:10}`. Even though the objects contain the same values, they are not the same object reference as each component creates its own `props` object.
+
+React will call [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) on the input to verify if there is a cache hit.
+
+```js {3,9}
+import {cache} from 'react';
+
+const calculateNorm = cache((x, y, z) => {
+  // ...
+});
+
+function MapMarker(props) {
+  // ✅ Good: Pass primitives to memoized function
+  const length = calculateNorm(props.x, props.y, props.z);
```

Comment:
Especially for the preload example, it seems more ergonomic to still use the `preload(props)` pattern. So as an alternative, your `calculateNorm` would be a wrapper that unpacks the props instead. That keeps the order of arguments next to the implementation and people don't have to recall the right order of arguments:


```js
const cachedCalculateNorm = cache((x, y, z) => {
  // ...
});
const calculateNorm = props => cachedCalculateNorm(props.x, props.y, props.z)
```

The additional implementation boilerplate may seem a bit annoying but this is probably nice to keep handy for people complaining about argument order. Especially for cached functions with a lot of spread out callsites. If you don't share the cached function across modules, it's probably not worth adding the "unpacking objects"-wrapper.

I don't think we need to include this pattern but probably good to keep in mind if people focus too much on being annoyed with recalling argument order.